docs/examples/tools/cloud_resource_detector/resource_detector_metrics.py

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python3
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import time
+
+from opentelemetry import metrics
+from opentelemetry.exporter.cloud_monitoring import (
+    CloudMonitoringMetricsExporter,
+)
+from opentelemetry.sdk.metrics import Counter, MeterProvider
+from opentelemetry.sdk.resources import get_aggregated_resources
+from opentelemetry.tools.resource_detector import GoogleCloudResourceDetector
+
+# MUST be run on a Google tool!
+# Detect resources from the environment
+resources = get_aggregated_resources([GoogleCloudResourceDetector()])
```

Comment:
Should this have `raise_on_error`?

Reply:
I don't believe so, since its required that this code be run in a Google tool. The only case where `raise_on_error` should be used is code could be run both on AWS and GCE, meaning both GoogleCloudResourceDetector and AWSResourceDetector would be present (and obviously one failing). Otherwise, the user should just not attempt detection.